src/openms_gui/source/VISUAL/LayerData.cpp

```diff
@@ -94,7 +94,8 @@ namespace OpenMS
     // Return if no valid peak layer attached
     if (getPeakData()->size() == 0 || type != LayerData::DT_PEAK) { return; }
 
-    MSSpectrum & spectrum = getCurrentSpectrum();
+    // TODO: check if this is correct?
+    MSSpectrum & spectrum = getCurrentSpectrumMuteable();
```

Comment:
@timosachsenberg can you check whether this should be `MSSpectrum & spectrum = getPeakDataMuteable()->getSpectrum(current_spectrum_);`